src/plugins/auto/auto_schedule.cpp

```diff
@@ -0,0 +1,571 @@
+        // it's necessary to wait the loading network threads to stop here.
+        _autoSContext->_plugin->executorManager()->clear("AutoDeviceAsyncLoad");
+        _executor.reset();
+    }
+    _autoSContext->_plugin->UnregisterPriority(_autoSContext->_modelPriority,
+        _loadContext[ACTUALDEVICE].deviceInfo.uniqueName);
+
+    LOG_INFO("[AUTOPLUGIN]ExecutableNetwork end");
+}
+
+
+IInferPtr AutoSchedule::CreateInferRequestImpl(
+    const std::vector<std::shared_ptr<const ov::Node>>& inputs,
+    const std::vector<std::shared_ptr<const ov::Node>>& outputs) {
+    auto num = _numRequestsCreated++;
+    IE::SoIInferRequestInternal request_to_share_blobs_with;
+    IE::RemoteContext::Ptr ctx = nullptr;
+    if (!_loadContext[CPU].isEnabled && _loadContext[ACTUALDEVICE].isAlready) {
+        try {
+            ctx = _autoSContext->_core->GetDefaultContext(
+                    _loadContext[ACTUALDEVICE].deviceInfo.deviceName);
+        } catch (IE::Exception& ex) {
+            // plugin does not support context, say CPU
+            LOG_DEBUG("[AUTOPLUGIN]context not supported for %s, fallback to default memory",
+                _loadContext[ACTUALDEVICE].deviceInfo.deviceName.c_str());
+            // for dynamic shape support
+            auto& dev_requests =
+                _workerRequests[_loadContext[ACTUALDEVICE].deviceInfo.deviceName];
+            if (num < dev_requests.size()) {
+                request_to_share_blobs_with = dev_requests.at(num)._inferRequest;
+            }
+        }
+    }
+    return std::make_shared<MultiDeviceInferRequest>(inputs, outputs, request_to_share_blobs_with, ctx);
+}
+
+IInferPtr AutoSchedule::CreateInferRequestImpl(IE::InputsDataMap networkInputs,
+    IE::OutputsDataMap networkOutputs) {
+    auto num = _numRequestsCreated++;
+    SoInfer request_to_share_blobs_with;
+    IE::RemoteContext::Ptr ctx = nullptr;
+    if (!_loadContext[CPU].isEnabled && _loadContext[ACTUALDEVICE].isAlready) {
+        try {
+            ctx = _autoSContext->_core->GetDefaultContext(
+                    _loadContext[ACTUALDEVICE].deviceInfo.deviceName);
+        } catch (IE::Exception& ex) {
+            // plugin does not support context
+            LOG_DEBUG("[AUTOPLUGIN]context not supported for %s, fallback to default memory",
+                _loadContext[ACTUALDEVICE].deviceInfo.deviceName.c_str());
+            auto& dev_requests =
+                _workerRequests[_loadContext[ACTUALDEVICE].deviceInfo.deviceName];
+            if (num < dev_requests.size()) {
+                request_to_share_blobs_with = dev_requests.at(num)._inferRequest;
+            }
+        }
+    }
+    return std::make_shared<MultiDeviceInferRequest>(networkInputs, networkOutputs, request_to_share_blobs_with, ctx);
+}
+
+IInferPtr AutoSchedule::CreateInferRequest() {
+    auto execNetwork = std::dynamic_pointer_cast<AutoExecutableNetwork>(
+            _autoSContext->_executableNetwork.lock());
+    if (_passthroughExeNet) {
```

Comment:
if will implement single schedule for case AUTO:singledevice, MULTI:singledevie, or some cases may cause single device, for exmple AUTO, in the future